contracts/bond/Redeemable.sol

```diff
@@ -41,14 +43,19 @@ abstract contract Redeemable is Initializable {
         return _redeemable;
     }
 
+    function reason() external view returns (string memory) {
+        return _reason;
+    }
+
     function __Redeemable_init() internal onlyInitializing {}
 
     /**
      * @dev Transitions redeemable from `false` to `true`.
      *
      * No affect if state is already transitioned.
      */
-    function _allowRedemption() internal {
+    function _allowRedemption(string memory reason) internal {
```

Comment:
question; can `calldata` be used here?

Reply:
it can indeed. had it in my head that internal functions must be memory for some reason!